src/cmp.rs

```diff
@@ -0,0 +1,1188 @@
+        executable,
+        ..Default::default()
+    };
+    let mut from = None;
+    let mut to = None;
+    let mut skip_pos1 = None;
+    let mut skip_pos2 = None;
+    while let Some(param) = opts.next() {
+        if param == "--" {
+            break;
+        }
+        if param == "-" {
+            if from.is_none() {
+                from = Some(param);
+            } else if to.is_none() {
+                to = Some(param);
+            } else {
+                return Err(usage_string(&executable_str));
+            }
+            continue;
+        }
+        if param == "-b" || param == "--print-bytes" {
+            params.print_bytes = true;
+            continue;
+        }
+        if param == "-l" || param == "--verbose" {
+            params.verbose = true;
+            continue;
+        }
+        if param == "-lb" || param == "-bl" {
+            params.print_bytes = true;
+            params.verbose = true;
+            continue;
+        }
+
+        let param_str = param.to_string_lossy().to_string();
+        if param == "-n" || param_str.starts_with("--bytes=") {
+            let max_bytes = if param == "-n" {
+                opts.next()
+                    .ok_or_else(|| usage_string(&executable_str))?
+                    .to_string_lossy()
+                    .to_string()
+            } else {
+                let (_, arg) = param_str.split_once('=').unwrap();
+                arg.to_string()
+            };
+            let max_bytes = match max_bytes.parse::<usize>() {
+                Ok(num) => num,
+                Err(e) if *e.kind() == std::num::IntErrorKind::PosOverflow => usize::MAX,
+                Err(_) => {
+                    return Err(format!(
+                        "{}: invalid --bytes value '{}'",
+                        executable_str, max_bytes
+                    ))
+                }
+            };
+            params.max_bytes = Some(max_bytes);
+            continue;
+        }
+        if param == "-i" || param_str.starts_with("--ignore-initial=") {
+            let skip_desc = if param == "-i" {
+                opts.next()
+                    .ok_or_else(|| usage_string(&executable_str))?
+                    .to_string_lossy()
+                    .to_string()
+            } else {
+                let (_, arg) = param_str.split_once('=').unwrap();
+                arg.to_string()
+            };
+            let (skip_a, skip_b) = if let Some((skip_a, skip_b)) = skip_desc.split_once(':') {
+                (
+                    parse_skip(skip_a, &skip_desc)?,
+                    parse_skip(skip_b, &skip_desc)?,
+                )
+            } else {
+                let skip = parse_skip(&skip_desc, &skip_desc)?;
+                (skip, skip)
+            };
+            params.skip_a = Some(skip_a);
+            params.skip_b = Some(skip_b);
+            continue;
+        }
+        if param == "-s" || param == "-q" || param == "--quiet" || param == "--silent" {
```

Comment:
`-q` doesn't appear to be a valid option for the GNU implementation. Is there a particular reason for accepting it here?